File: Data Structures & Algorithms/merge-two-sorted-linked-lists/submission-0.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        if (list1 == null) return list2;
        if (list2 == null) return list1;

        ListNode list3 = null, cur3 = null;
        ListNode cur1 = list1, cur2 = list2;
        while (cur1 != null || cur2 != null) {
            if (cur3 == null) {
                cur3 = new ListNode();
                list3 = cur3;
            } else {
                cur3.next = new ListNode();
                cur3 = cur3.next;
            }

            if (cur1 == null) {
                cur3.val = cur2.val;
                cur2 = cur2.next; 
            } else if (cur2 == null) {
                cur3.val = cur1.val;
                cur1 = cur1.next;
            } else {
                if (cur1.val <= cur2.val) {
                    cur3.val = cur1.val;
                    cur1 = cur1.next;
                } else if (cur2.val < cur1.val) {
                    cur3.val = cur2.val;
                    cur2 = cur2.next;
                }
            }
        }

        return list3;
    }
}

File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        # First find the length of the list
        length = 0
        cur = head
        while cur:
            length += 1
            cur = cur.next

        # Now that we know the length of the list, 
        # we use it to find the n'th node from the end
        # which is the same as the (length - n)th node 
        # from the front
        dummy = ListNode(-1,head)
        cur = dummy
        for _ in range(length - n):
            cur = cur.next

        # Sever the node from the list
        cur.next = cur.next.next

        return dummy.next


        